java.lang.IllegalArgumentException: Failure in creating proxy URL. Proxy port is required!

## Description

From https://github.com/quarkusio/quarkus/discussions/37137

It seems that environment variables such as:

```
ALL_PROXY=proxy-se-uan.ddc.teliasonera.net:8080
NO_PROXY=localhost
```

Are causing the following stack-trace:
``` 
 java.base/java.lang.Thread.run(Thread.java:1583) Caused by: java.lang.IllegalArgumentException: Failure in creating proxy URL. Proxy port is required! at
   io.fabric8.kubernetes.client.utils.HttpClientUtils.getProxyUri(HttpClientUtils.java:91) at
   io.fabric8.kubernetes.client.utils.HttpClientUtils.configureProxy(HttpClientUtils.java:210) at
   io.fabric8.kubernetes.client.utils.HttpClientUtils.applyCommonConfiguration(HttpClientUtils.java:185) at io.fabric8.kubernetes.client.http.HttpClient$Factory.newBuilder(HttpClient.java:51) at
   io.quarkus.kubernetes.client.runtime.QuarkusHttpClientFactory.newBuilder(QuarkusHttpClientFactory.java:57) at
   io.fabric8.kubernetes.client.KubernetesClientBuilder.getHttpClient(KubernetesClientBuilder.java:90) at io.fabric8.kubernetes.client.KubernetesClientBuilder.build(KubernetesClientBuilder.java:79) at
   io.quarkus.kubernetes.client.runtime.KubernetesClientUtils.createClient(KubernetesClientUtils.java:54) at
   io.quarkus.kubernetes.config.runtime.KubernetesConfigSourceFactoryBuilder$KubernetesConfigFactory.getConfigSources(KubernetesConfigSourceFactoryBuilder.java:31) at ....

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 * Fix #5866: Addressed cycle in crd generation with Java 19+ and ZonedDateTime
 
 #### Improvements
+* Fix #5605: proxy configs support proxy URLs without a scheme
 * Fix #5878: (java-generator) Add implements Editable for extraAnnotations
 * Fix #5878: (java-generator) Update documentation to include dependencies
 * Fix #5867: (crd-generator) Imply schemaFrom via JsonFormat shape (SchemaFrom takes precedence)

--- a/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java
+++ b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java
@@ -86,7 +86,14 @@ public class HttpClientUtils {
       proxy = config.getHttpProxy();
     }
     if (proxy != null) {
-      URI proxyUrl = new URI(proxy);
+      final String completedProxy;
+      if (proxy.contains("://")) {
+        completedProxy = proxy;
+      } else {
+        // No protocol specified, default to cluster requirements
+        completedProxy = master.getProtocol() + "://" + proxy;
+      }
+      final URI proxyUrl = new URI(completedProxy);
       if (proxyUrl.getPort() < 0) {
         throw new IllegalArgumentException("Failure in creating proxy URL. Proxy port is required!");
       }

--- a/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/utils/HttpClientUtilsTest.java
+++ b/kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/utils/HttpClientUtilsTest.java
@@ -65,12 +65,12 @@ import static org.mockito.Mockito.when;
 class HttpClientUtilsTest {
 
   @Test
-  void toProxyTypeTestUnknown() throws MalformedURLException {
+  void toProxyTypeTestUnknown() {
     assertThrows(MalformedURLException.class, () -> HttpClientUtils.toProxyType("unknown"));
   }
 
   @Test
-  void toProxyTypeTestNull() throws MalformedURLException {
+  void toProxyTypeTestNull() {
     assertThrows(MalformedURLException.class, () -> HttpClientUtils.toProxyType(null));
   }
 
@@ -160,7 +160,7 @@ class HttpClientUtilsTest {
     @ParameterizedTest(name = "{index}: Master hostname ''{0}'' matched by No Proxy ''{1}'' ")
     @MethodSource("masterHostnameDoesMatchNoProxyInput")
     void masterHostnameDoesMatchNoProxy(String masterHostname, String[] noProxy)
-        throws MalformedURLException, URISyntaxException {
+        throws MalformedURLException {
       assertTrue(HttpClientUtils.isHostMatchedByNoProxy(masterHostname, noProxy));
     }
 
@@ -235,6 +235,19 @@ class HttpClientUtilsTest {
       URI url = HttpClientUtils.getProxyUri(new URL("http://localhost"), config);
       // Then
       assertThat(url).isNull();
+    }
+
+    @Test
+    void whenIncompleteHttpProxyUrlProvided_shouldInferProtocol() throws MalformedURLException, URISyntaxException {
+      // Given
+      Config config = configBuilder.withHttpProxy("example.com:8080").build();
+      // When
+      URI url = HttpClientUtils.getProxyUri(new URL("http://localhost"), config);
+      // Then
+      assertThat(url)
+          .hasScheme("http")
+          .hasHost("example.com")
+          .hasPort(8080);
     }
   }
 

